tests/Aspire.Hosting.Tests/MSBuildTests.cs

```diff
@@ -124,4 +123,132 @@ the Aspire.AppHost.SDK targets that will automatically add these references to p
         Assert.Contains("warning ASPIRE004", output.ToString());
     }
+
+    /// <summary>
+    /// Tests that the metadata sources are emitted correctly.
+    /// </summary>
+    [Fact]
+    public async Task ValidateMetadataSources()
+    {
+        var repoRoot = MSBuildUtils.GetRepoRoot();
+        using var tempDirectory = new TempDirectory();
+
+        var appDirectory = Path.Combine(tempDirectory.Path, "App");
+        Directory.CreateDirectory(appDirectory);
+
+        File.WriteAllText(Path.Combine(appDirectory, "App.csproj"), """
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+</Project>
+""");
+        File.WriteAllText(Path.Combine(appDirectory, "Program.cs"), """
+Console.WriteLine("Hello, Aspire!");
+""");
+
+        var appHostDirectory = Path.Combine(tempDirectory.Path, "AppHost");
+        Directory.CreateDirectory(appHostDirectory);
+
+        File.WriteAllText(Path.Combine(appHostDirectory, "AppHost.csproj"), $"""
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+    <IsAspireHost>true</IsAspireHost>
+
+    <!-- 
+      Test applications have their own way of referencing Aspire.Hosting.AppHost, as well as DCP and Dashboard, so we disable
+      the Aspire.AppHost.SDK targets that will automatically add these references to projects. 
+    -->
+    <SkipAddAspireDefaultReferences Condition="'$(TestsRunningOutsideOfRepo)' != 'true'">true</SkipAddAspireDefaultReferences>
+    <AspireHostingSDKVersion>9.0.0</AspireHostingSDKVersion>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="{repoRoot}\src\Aspire.Hosting.AppHost\Aspire.Hosting.AppHost.csproj" IsAspireProjectResource="false" />
+    <ProjectReference Include="..\App\App.csproj" />
+  </ItemGroup>
+
+</Project>
+""");
+        File.WriteAllText(Path.Combine(appHostDirectory, "AppHost.cs"), """
+var builder = DistributedApplication.CreateBuilder();
+builder.Build().Run();
+""");
+
+        File.WriteAllText(Path.Combine(appHostDirectory, "Directory.Build.props"), $"""
+<Project>
+  <PropertyGroup>
+    <SkipAspireWorkloadManifest>true</SkipAspireWorkloadManifest>
+  </PropertyGroup>
+
+  <Import Project="{repoRoot}\src\Aspire.Hosting.AppHost\build\Aspire.Hosting.AppHost.props" />
+</Project>
+""");
+        File.WriteAllText(Path.Combine(appHostDirectory, "Directory.Build.targets"), $"""
+<Project>
+  <Import Project="{repoRoot}\src\Aspire.Hosting.AppHost\build\Aspire.Hosting.AppHost.in.targets" />
+  <Import Project="{repoRoot}\src\Aspire.AppHost.Sdk\SDK\Sdk.in.targets" />
+</Project>
+""");
+
+        var binlogPath = Path.Join(repoRoot, "artifacts", "log", "Debug", $"{nameof(ValidateMetadataSources)}.binlog");
```

Comment:
I don't think we need the binlog anymore.

Reply:
OK, I thought it would be nice for the next person to debug, but I'll remove for now.

Reply:
Removed!